app/models/pin.rb

```diff
@@ -1,2 +1,6 @@
 class Pin < ActiveRecord::Base
+  validates :pininterest_id, uniqueness: true
+  validates :pin_url,  presence: true #this is the original src url
```

Comment:
this is the original url for the pin - not the pinterest url. let's call it...... source_url??? will change